### PR TITLE
Minor apcs_list refactor

### PR DESCRIPTION
**Pull request by WarlockD**
## About The Pull Request

So been slowly doing a machinery refactor but just can't do that without a powernet refactor so maybe do it all in small prs that piss me off.  So right now going to do a bunch of small pr's about removing liner searches in global lists.  All this does is change the apc global list to an associated lists with area keys.

## Why It's Good For The Game

Binary searches are bad! 

## Changelog
:cl:
refactor: GLOB.apcs_list 
/:cl:

### CODE REVIEW

**Comment by Cyberboss** [comment]
What happens if an area has multiple APCs (I forget if this is possible)

**Comment by WarlockD** [comment]
Nah its not.  The refactor I am doing is moving all power to powernets so there won't be a difference between using wire or an apc as long as the machine has the powernet.  So in theory you could set up APC's in area less areas and do it based on range.

File: code/_globalvars/lists/objects.dm [hunk 1]
```diff
@@ -22,7 +22,7 @@ GLOBAL_LIST_EMPTY(tech_list)					//list of all /datum/tech datums indexed by id.
 GLOBAL_LIST_EMPTY(surgeries_list)				//list of all surgeries by name, associated with their path.
 GLOBAL_LIST_EMPTY(crafting_recipes)				//list of all table craft recipes
 GLOBAL_LIST_EMPTY(rcd_list)					//list of Rapid Construction Devices.
-GLOBAL_LIST_EMPTY(apcs_list)					//list of all Area Power Controller machines, separate from machines for powernet speeeeeeed.
+GLOBAL_LIST_EMPTY(apcs_list)					//list of all Area Power Controller machines, separate from machines for powernet speeeeeeed. Associated list [AREA] = APC
```
Comment (MrPerson) [inline code/_globalvars/lists/objects.dm:25]: You should completely replace the previous comment. This is NOT a list of area power controller machines!

**Comment by MrPerson** [comment]
I know this isn't really what you want to be doing, but for the record, this list is unnecessary. All of its uses could be replaced with either a reference /area/var/my_apc or a global signal to communicate with any/all APCs.

**Comment by WarlockD** [comment]
> I know this isn't really what you want to be doing, but for the record, this list is unnecessary. All of its uses could be replaced with either a reference /area/var/my_apc or a global signal to communicate with any/all APCs.

Your right.  I have been flip floping between removing the list and just making a var in area and keeping this.  Humm.  Ok let me switch it around and see if I can just get rid of the list.

**Comment by WarlockD** [comment]
Humm...damnit.  Here I was thinking this is a simple pr, but if I put the apc in area, I should refactor the power flags as well.  Lets see if I can keep this pr narrow.

File: code/_globalvars/lists/objects.dm (hunk 1 above)
Comment (WarlockD) [reply code/_globalvars/lists/objects.dm:25]: Oh good point I just wanted to message the format

**Comment by WarlockD** [comment]
I am going to hold off on moving the apc to the area for now.  I don't want to just add a var and ignore the other issues with power in areas (staticpower, the other area vars that aren't needed if the apc ref is in the area) as that is all going into a powernet/machinery revamp I am working on.

I am also working on a ntnet/signal system that would remove the need for alot of global vars.

**Comment by WarlockD** [comment]
uggh wrong revert

**Comment by MrPerson** [comment]
As long as the code is better, it doesn't really matter if you refactor power or machines or both or neither and do something else entirely. Anything you do is appreciated.

**Comment by WarlockD** [comment]
Ok I just spent 7 hours finishing my network remake, going to put this on hold and slow down a bit

**Comment by Dennok** [comment]
> What happens if an area has multiple APCs (I forget if this is possible)

One area must have one APC
Now if area has 2 APC and one of APC upowers, it disable power in area despite the fact that the second APC is full charged

**Pull request closed without merging**